ci/common.rb

```diff
@@ -178,6 +178,29 @@ def self.for(smth, max_timeout = DEFAULT_TIMEOUT)
       sh %(#{path}nosetests -s -v -A "#{nose}" #{tests_directory})
       t.reenable
     end
-    task execute: [:before_install, :install, :before_script, :script]
+
+    task :execute, :flavor do |_t, attr|
+      flavor = attr[:flavor]
+      # Check if the check uses the cache
+      cached = flavor.tasks.any? { |task| task.name == (flavor.scope.path + ':before_cache') }
```

Comment:
We should proably double check flavors that have the `before_cache` tasks but actually don't use it.
I think this is why Win tests are failing during the `before_cache` task  for `core_integrations`: it was not executed before but now it is since it exists and `cached` becomes true.
Haven't had the opportunity to verify my thoughts, let me know if this makes sense.